### PR TITLE
lazy load azure credentials for app and fix cache ctx cancel

### PR DESCRIPTION
**Pull request by greedy52**

### CODE REVIEW

File: lib/srv/app/azure/handler.go [hunk 1]
```diff
@@ -308,6 +304,8 @@ func (s *handler) getToken(ctx context.Context, managedIdentity string, scope st
 	select {
 	case <-timeoutChan:
 		return nil, trace.Wrap(context.DeadlineExceeded, "timeout waiting for access token for %v", getTokenTimeout)
+	case <-ctx.Done():
+		return nil, ctx.Err()
 	case <-cancelCtx.Done():
 		return tokenResult, errorResult
```
Comment (greedy52) [inline lib/srv/app/azure/handler.go:310]: caught by gavin that this line can return `nil, nil` if `ctx` is canceled

Comment (greedy52) [reply lib/srv/app/azure/handler.go:310]: my first attempt is still racy, let me fix it with a different chan

**Review by Tener — APPROVED** [review]
Looks good to me. What was the original issue that gets fixed here?

**Review by GavinFrazar — APPROVED** [review]

**Comment by Tener** [comment]
I think this one is good for merging?

**Review by smallinsky — APPROVED** [review]

**Pull request merged**